flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2245,6 +2245,31 @@ class ArrayExprLowering {
     ael.lowerArrayAssignment(lhs, rhs);
   }
 
+  /// Entry point for assignment to allocatable array.
+  static void lowerAllocatableArrayAssignment(
+      Fortran::lower::AbstractConverter &converter,
+      Fortran::lower::SymMap &symMap, Fortran::lower::StatementContext &stmtCtx,
+      const fir::MutableBoxValue &lhs,
+      const Fortran::evaluate::Expr<Fortran::evaluate::SomeType> &rhs) {
+    // If lbounds(expr) could be different than `1`, the allocatable must
+    // take lower bounds from the expr if reallocated.
+    auto takesLbound =
```

Comment:
Does this mean "might take"? Or do we always impose an lbound value whenever the rhs is a symbol or component path?

Reply:
This read might, it is lbounds are only set if the entity is reallocated and the the rhs has lbounds. Renamed to `takeLboundsIfRealloc` and clarified comment.